### PR TITLE
Refresh page after adding/editing team member

### LINKED ISSUE #1741: Refresh page after adding/editing team member
I have to refresh the page to see the newly added or edited team member details on the team's page.

### PR DESCRIPTION
**Pull request by Shruti-Apte**
Closes #1741 

### What 
- Page refresh after add/edit/delete
- Compensation tab hidden from profile and team details
- Employment and devices tab added on team details

### CODE REVIEW

File: app/javascript/src/components/Team/modals/AddEditMember.tsx [hunk 1]
```diff
@@ -72,13 +72,17 @@ const EditClient = ({ user = {}, isEdit = false }: Props) => {
       if (isEdit) {
         if (user.isTeamMember) {
           await teamApi.updateTeamMember(user.id, payload);
+          setRefreshList(true);
```
Comment (prasanthchaduvula) [inline app/javascript/src/components/Team/modals/AddEditMember.tsx:75]: ditto

File: app/javascript/src/components/Team/modals/DeleteMember.tsx [hunk 2]
```diff
@@ -11,14 +11,16 @@ import { useList } from "context/TeamContext";
...
+  const { setModalState, modal, setRefreshList } = useList();
 
   const deleteTeamMember = async () => {
     try {
       if (user.isTeamMember) {
         await teamApi.destroyTeamMember(user.id);
+        setRefreshList(true);
       } else {
         await teamApi.deleteInvitedMember(user.id);
+        setRefreshList(true);
```
Comment (prasanthchaduvula) [inline app/javascript/src/components/Team/modals/DeleteMember.tsx:23]: ditto

**Review by prasanthchaduvula — CHANGES_REQUESTED** [review]

**Comment by apoorv1316** [comment]
@prasanthchaduvula Please review again

File: app/javascript/src/components/Team/modals/AddEditMember.tsx [hunk 3]
```diff
@@ -71,12 +71,39 @@ const EditClient = ({ user = {}, isEdit = false }: Props) => {
...
+              return updatedUser;
+            }
+
+            return member;
+          });
+          setTeamList(updatedTeamList);
         } else {
-          await teamApi.updateInvitedMember(user.id, payload);
+          const res = await teamApi.updateInvitedMember(user.id, payload);
+          const updatedUser = res.data.invitation;
```
Comment (prasanthchaduvula) [inline app/javascript/src/components/Team/modals/AddEditMember.tsx:86]: Create `updatedTeamList` function. Extract the below logic and call it inside the above if and this else conditions and pass the` res.data.user `or `res.data.invitation` as param.
```
teamList.map(member => {
            if (member.id === updatedUser.id) {
              return updatedUser;
            }

            return member;
          });
          setTeamList(updatedTeamList);
```

File: app/javascript/src/components/Team/modals/DeleteMember.tsx [hunk 4]
```diff
@@ -11,14 +11,22 @@ import { useList } from "context/TeamContext";
...
-        await teamApi.destroyTeamMember(user.id);
+        const res = await teamApi.destroyTeamMember(user.id);
+        const updatedTeamList = teamList.filter(
+          member => member.id !== res.data.user.id
+        );
+        setTeamList(updatedTeamList);
       } else {
-        await teamApi.deleteInvitedMember(user.id);
+        const res = await teamApi.deleteInvitedMember(user.id);
+        const updatedTeamList = teamList.filter(
```
Comment (prasanthchaduvula) [inline app/javascript/src/components/Team/modals/DeleteMember.tsx:26]: create one function named `updatedTeamListAferDelete` with below code and pass `res.data.user.id` or `res.data.id` and call it inside the if and else conditions

```
teamList.filter(
          member => member.id !== id
        );
        setTeamList(updatedTeamList);
```

**Review by prasanthchaduvula — CHANGES_REQUESTED** [review]

**Review by prasanthchaduvula — APPROVED** [review]

**Pull request merged**